vsintegration/src/FSharp.Editor/Common/Extensions.fs

```diff
@@ -24,6 +25,34 @@ type System.IServiceProvider with
     member x.GetService<'T>() = x.GetService(typeof<'T>) :?> 'T
     member x.GetService<'S, 'T>() = x.GetService(typeof<'S>) :?> 'T
 
+type Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem.AbstractProject with
+
+    member this.GetCurrentProjectReferenceIds() =
+        this.GetCurrentProjectReferences() 
+        |> Seq.map (fun x -> x.ProjectId) 
+        |> ImmutableArray.ToImmutableArray
+
+type Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem.VisualStudioWorkspaceImpl with
+
+    member this.GetCurrentProjectReferenceIds(projectId: ProjectId) =
+        match this.ProjectTracker.GetProject(projectId) with
```

Comment:
Do you need to use the method that will cause the project's deferred state to be created here if it isn't already?

Reply:
I don't think so if I'm understanding you correctly.